train.py

```diff
@@ -0,0 +1,153 @@
+import hydra
+import torch
+import numpy as np
+from tqdm import tqdm
+from torch import nn, optim
+from torch.utils.data import DataLoader
+from omegaconf import OmegaConf, DictConfig
+
+import wandb
+from model.VQVAE import VQVAE
+from utils.data_loader import create_loaders
+
+
+def count_parameters(model):
+    # https://discuss.pytorch.org/t/how-do-i-check-the-number-of-parameters-of-a-model/4325/9
+    print(f"The model has {sum(p.numel() for p in model.parameters() if p.requires_grad):,} trainable parameters.")
+
+
+def set_seed(seed):
+    # https://pytorch.org/docs/stable/notes/randomness.html
+    torch.manual_seed(seed)
+    torch.cuda.manual_seed_all(seed)
+    np.random.seed(seed)
+    torch.backends.cudnn.deterministic = True
+    torch.backends.cudnn.benchmark = False
+
+
+def test(model: VQVAE, test_loader: DataLoader, criterion, cfg: DictConfig, epoch: int):
+    model.eval()
+    total_test_loss = 0
+    total_reconstruction_loss = 0
+    n_test = 0
+
+    with torch.no_grad():
+        for batch in test_loader:
+            x_combined = torch.stack([batch["start"], batch["duration"], batch["velocity"]], dim=1)
+
+            x_combined = x_combined.to(cfg.system.device)
+
+            reconstructed_x, vq_loss, losses, perplexity, _, _ = model(x_combined)
+
+            # Compute the loss
+            reconstruction_loss = criterion(reconstructed_x, x_combined)
+            total_reconstruction_loss += reconstruction_loss.item()
+
+            loss = reconstruction_loss + vq_loss
+            total_test_loss += loss.item()
+            n_test += 1
+
+        avg_test_loss = total_test_loss / n_test
+        avg_reconstruction_loss = total_reconstruction_loss / n_test
+
+        # Log the test metrics to wandb
+        wandb.log(
+            {
+                "test/Average Test Loss": avg_test_loss,
+                "test/Reconstruction Loss": avg_reconstruction_loss,
+            }
+        )
+        print(f"Epoch {epoch}, Average Test Loss: {avg_test_loss}")
+
+
+def train(model: VQVAE, train_loader: DataLoader, optimizer, criterion, cfg: DictConfig, epoch: int):
```

Comment:
```suggestion
def train_step(model: VQVAE, train_loader: DataLoader, optimizer, criterion, cfg: DictConfig, epoch: int):
```

`train` shows up a lot as a string, and grepping around becomes difficult :)